azurerm/resource_arm_sql_firewall_rule.go

```diff
@@ -51,96 +50,88 @@ func resourceArmSqlFirewallRule() *schema.Resource {
+	endIpAddress := d.Get("end_ip_address").(string)
+
+	props := sql.FirewallRuleProperties{
+		StartIPAddress: &startIpAddress,
+		EndIPAddress:   &endIpAddress,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &sql.GetFirewallRule{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ServerName:        d.Get("server_name").(string),
+	parameters := sql.FirewallRule{
+		Name: &name,
+		FirewallRuleProperties: &props,
 	}
 
-	readResponse, err := readRequest.Execute()
+	//last parameter is set to empty to allow updates to records after creation
+	// (per SDK, set it to '*' to prevent updates, all other values are ignored)
+	result, err := sqlFirewallRulesClient.CreateOrUpdate(resGroup, serverName, name, parameters)
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Server Firewall Rule: %s", err)
+		return err
 	}
-	if !readResponse.IsSuccessful() {
-		return fmt.Errorf("Error reading SQL Server Firewall Rule: %s", readResponse.Error)
+
+	if result.ID == nil {
+		return fmt.Errorf("Cannot create sql firewall rule %s (resource group %s) ID", name, resGroup)
 	}
 
-	resp := readResponse.Parsed.(*sql.GetFirewallRuleResponse)
-	d.SetId(*resp.ID)
+	d.SetId(*result.ID)
 
 	return resourceArmSqlFirewallRuleRead(d, meta)
 }
 
 func resourceArmSqlFirewallRuleRead(d *schema.ResourceData, meta interface{}) error {
+	sqlFirewallRulesClient := meta.(*ArmClient).sqlFirewallRulesClient
+
 	id, err := parseAzureResourceID(d.Id())
 	if err != nil {
 		return err
 	}
-	resGroup := id.ResourceGroup
 
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
-
-	readRequest := rivieraClient.NewRequestForURI(d.Id())
-	readRequest.Command = &sql.GetFirewallRule{}
+	resGroup := id.ResourceGroup
+	name := id.Path["firewallRules"]
+	serverName := id.Path["servers"]
 
-	readResponse, err := readRequest.Execute()
+	result, err := sqlFirewallRulesClient.Get(resGroup, serverName, name)
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Server Firewall Rule: %s", err)
+		return fmt.Errorf("Error reading DNS PTR record %s: %v", name, err)
 	}
-	if !readResponse.IsSuccessful() {
-		log.Printf("[INFO] Error reading SQL Server Firewall Rule %q - removing from state", d.Id())
+	if result.Response.StatusCode == http.StatusNotFound {
 		d.SetId("")
-		return fmt.Errorf("Error reading SQL Server Firewall Rule: %s", readResponse.Error)
+		return nil
 	}
 
-	resp := readResponse.Parsed.(*sql.GetFirewallRuleResponse)
+	props := *result.FirewallRuleProperties
 
+	d.Set("name", name)
 	d.Set("resource_group_name", resGroup)
-	d.Set("location", azureRMNormalizeLocation(*resp.Location))
-	d.Set("name", resp.Name)
-	d.Set("server_name", id.Path["servers"])
-	d.Set("start_ip_address", resp.StartIPAddress)
-	d.Set("end_ip_address", resp.EndIPAddress)
+	d.Set("server_name", serverName)
+	d.Set("start_ip_address", *props.StartIPAddress)
+	d.Set("end_ip_address", *props.EndIPAddress)
```

Comment:
can we add a check to ensure props isn't `nil` around these properties (for invalid Swagger/to cover a bad api responses) i.e.

```
if props := result.FirewallRuleProperties; props != nil {
  d.Set("start_ip_address", props.StartIPAddress)
 ...
}
```

(as mentioned in the other PR, we shouldn't need the `*` here)